resources/innosetup/setup.iss

```diff
@@ -57,15 +57,31 @@ Name: "webdav"; Description: "WebDAV system configuration"; Types: full; ExtraDi
-Root: HKCR; Subkey: ".cryptomator"; ValueType: string;  ValueName: ""; ValueData: "CryptomatorMasterkeyFile"; Flags: uninsdeletekey
-Root: HKCR; Subkey: "CryptomatorMasterkeyFile"; ValueType: string;  ValueName: ""; ValueData: "Cryptomator Masterkey File"; Flags: uninsdeletekey
-Root: HKCR; Subkey: "CryptomatorMasterkeyFile\DefaultIcon"; ValueType: string; ValueName: ""; ValueData: "{app}\Cryptomator.exe,0"
-Root: HKCR; Subkey: "CryptomatorMasterkeyFile\shell\open\command"; ValueType: string; ValueName: ""; ValueData: """{app}\Cryptomator.exe"" ""%1"""
-Root: HKCR; Subkey: ".c9r"; ValueType: string;  ValueName: ""; ValueData: "EncryptedCryptomatorVaultFile"; Flags: uninsdeletekey
-Root: HKCR; Subkey: "EncryptedCryptomatorVaultFile"; ValueType: string;  ValueName: ""; ValueData: "Encrypted Cryptomator vault file"; Flags: uninsdeletekey
-Root: HKCR; Subkey: "EncryptedCryptomatorVaultFile\DefaultIcon"; ValueType: string; ValueName: ""; ValueData: "{app}\Cryptomator.exe,0"
+
+; Remove legacy AutoStart entry; see: https://github.com/cryptomator/cryptomator-win/issues/33
 Root: HKCU; Subkey: "SOFTWARE\Microsoft\Windows\CurrentVersion\Run\"; ValueType: none; ValueName: "Cryptomator"; Flags: dontcreatekey deletevalue
 
+; .cryptomator filetype: Add extension, MIME type, perceived type, description, icon and command
+Root: HKA; Subkey: "Software\Classes\.cryptomator"; ValueType: string;  ValueName: ""; ValueData: "CryptomatorMasterkeyFile"; Flags: uninsdeletekey
+Root: HKA; Subkey: "Software\Classes\.cryptomator"; ValueType: string; ValueName: "Content Type"; ValueData: "application/x-vnd.cryptomator.vault-metadata"
+Root: HKA; Subkey: "Software\Classes\.cryptomator"; ValueType: string; ValueName: "PerceivedType"; ValueData: "text"
+
+Root: HKA; Subkey: "Software\Classes\CryptomatorMasterkeyFile"; ValueType: string;  ValueName: ""; ValueData: "Cryptomator Masterkey File"; Flags: uninsdeletekey
+Root: HKA; Subkey: "Software\Classes\CryptomatorMasterkeyFile\DefaultIcon"; ValueType: string; ValueName: ""; ValueData: "{app}\Cryptomator.exe,0"
+Root: HKA; Subkey: "Software\Classes\CryptomatorMasterkeyFile\shell\open\command"; ValueType: string; ValueName: ""; ValueData: """{app}\Cryptomator.exe"" ""%1"""
+
+; .c9r & .c9s filetypes: Add extension, MIME type, perceived type, description and icon for each
+Root: HKA; Subkey: "Software\Classes\.c9r"; ValueType: string;  ValueName: ""; ValueData: "CryptomatorEncryptedData"; Flags: uninsdeletekey
+Root: HKA; Subkey: "Software\Classes\.c9r"; ValueType: string; ValueName: "Content Type"; ValueData: "application/x-vnd.cryptomator.encrypted-data"
+Root: HKA; Subkey: "Software\Classes\.c9r"; ValueType: string; ValueName: "PerceivedType"; ValueData: "system"
+
+Root: HKA; Subkey: "Software\Classes\.c9s"; ValueType: string;  ValueName: ""; ValueData: "CryptomatorEncryptedData"; Flags: uninsdeletekey
```

Comment:
Is there any official documentation confirming that it is allowed and intended to use the same name (`CryptomatorEncryptedData`) multiple times for different file extensions?

Reply:
Yes, indeed. See: [Here](https://web.archive.org/web/20210201153726/https://docs.microsoft.com/de-de/windows/win32/shell/how-to-register-a-file-type-for-a-new-application#step-1)